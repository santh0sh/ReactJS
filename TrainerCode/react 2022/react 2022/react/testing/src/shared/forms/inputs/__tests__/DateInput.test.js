import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import TestDateInput from '../DateInput';

const moment = require.requireActual('moment-timezone').tz.setDefault('America/Los_Angeles'),
    defaultProps = {
        minDate: moment(0)
    },
    DateInput = (props) =>
        <TestDateInput
            {...defaultProps}
            {...props}
        />;

describe('Render DateInput', () => {
    it('render correctly date component', () => {
        const DateInputComponent = renderer.create(<DateInput />).toJSON();
        expect(DateInputComponent).toMatchSnapshot();
    });

    it('render date input correctly with empty value', () => {
        const props = {
                value: null
            },
            DateInputComponent = mount(<DateInput {...props} />);
        expect((DateInputComponent).prop('value')).toEqual(null);
    });

    it('check the onChange callback', () => {
        const onChange = jest.fn(),
            props = {
                value: '20.01.2018',
                onChange
            },
            DateInputComponent = mount(<DateInput {...props} />).find('input');
        DateInputComponent.simulate('change', { target: {value: moment('2018-01-22')} });
        expect(onChange).toHaveBeenCalledWith('22.01.2018');
    });

    it('check the type of value', () => {
        const props = {
                value: '10.03.2018'
            },
            DateInputComponent = mount(<DateInput {...props} />);
        expect(DateInputComponent.prop('value')).toBeString();
    });

    it('check DatePicker popup open', () => {
        const DateComponent = mount(<DateInput />),
            dateInput = DateComponent.find("input[type='text']");
        dateInput.simulate('click');
        expect(DateComponent.find('.react-datepicker')).toHaveLength(1);
    });

    it('check month and years dropdowns displayed', () => {
        const props = {
                showMonthYearsDropdowns: true
            },
            DateInputComponent = mount(<DateInput {...props} />).find('.datepicker');
        expect(DateInputComponent.hasClass('react-datepicker-hide-month')).toEqual(true);
    });
});
