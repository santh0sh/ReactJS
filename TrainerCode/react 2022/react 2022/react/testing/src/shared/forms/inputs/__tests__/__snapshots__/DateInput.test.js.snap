// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render DateInput render correctly date component 1`] = `
<div
  className="datepicker "
>
  <div>
    <div
      className="react-datepicker-wrapper"
    >
      <div
        className="react-datepicker__input-container"
      >
        <input
          autoComplete={undefined}
          autoFocus={undefined}
          className="input-custom"
          disabled={false}
          id={undefined}
          name={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder={undefined}
          readOnly={undefined}
          required={undefined}
          tabIndex={undefined}
          title={undefined}
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
